test/zzz-rsyslog.sh

```diff
@@ -346,11 +346,11 @@ cat > $cfgdir/66-debug.conf << EOF
 # DEBUGGING
 set \$!undefined1 = "undefined1";
 set \$!undefined11 = 1111;
-set \$!undefined12 = "True";
+set \$!undefined12 = True;
```

Comment:
https://www.rsyslog.com/doc/v8-stable/rainerscript/data_types.html
```
RainerScript is a typeless language. That doesn’t imply you don’t need to care about types. Of course, expressions like “A” + “B” will not return a valid result, as you can’t really add two letters (to concatenate them, use the concatenation operator &).  However, all type conversions are automatically done by the script interpreter when there is need to do so.
```
also https://www.rsyslog.com/doc/v8-stable/rainerscript/variable_property_types.html

So I don't think you can set a JSON boolean value in rainierscript.  I also don't see how you can set an array valued object - just strings and numbers, and JSON hash via using the `$!hash!value` construction.

This begs the question of how the JSON parser in rsyslog works - how does it internally represent JSON boolean and JSON array values?

I think for testing purposes we'll need to figure out how to pass in JSON booleans and arrays.  I think we may have to have rsyslog parse a file, or use the `parse_json` function if it is available in this version of rsyslog.

Reply:
Looks promising.  I see `doFunc_parse_json` in origin-aggregated-logging/rsyslog/vendored_src/rsyslog/rsyslog/grammar/rainerscript.c.  I'll give it a try...